neuralop/layers/neighbor_search.py

```diff
@@ -61,31 +69,34 @@ def forward(self, data, queries, radius):
             return_dict['neighbors_row_splits'] = search_return.neighbors_row_splits.long()
 
         else:
-            neighbors_count, neighbors_index = self.search_fn(data, queries, radius,
-                                               max_num_neighbors=data.shape[0])
-            
-            if neighbors_count[-1] != queries.shape[0] - 1:
-                add_max_element = True
-                neighbors_count = torch.cat((neighbors_count,
-                                             torch.tensor([queries.shape[0] - 1], 
-                                                          dtype=neighbors_count.dtype,
-                                                          device=neighbors_count.device)), 
-                                                          dim=0)
+            if not self.use_torch_cluster:
+                return_dict = self.search_fn(data, queries, radius)
             else:
-                add_max_element = False
-            
-            bins = torch.bincount(neighbors_count, minlength=1)
-            if add_max_element:
-                bins[-1] -= 1
-            
-            neighbors_row_splits = torch.cumsum(bins, dim=0)
-            neighbors_row_splits = torch.cat((torch.tensor([0], 
-                                              dtype=neighbors_row_splits.dtype,
-                                              device=neighbors_row_splits.device),
-                                              neighbors_row_splits), dim=0)
-            
+                neighbors_count, neighbors_index = self.search_fn(data, queries, radius,
+                                                max_num_neighbors=data.shape[0])
```

Comment:
Should we wrap that code blow in a small, tested function?

Reply:
Added a unit test for fallback search

Reply:
Torch_cluster is slower than my implementation anyway. What if we just axed that coed